bee-node/src/main.rs

```diff
@@ -10,6 +9,35 @@ use log::error;
 
 const CONFIG_PATH: &str = "./config.toml";
 
+#[cfg(feature = "console")]
+fn logger_init() -> tokio::task::JoinHandle<Result<(), Box<dyn std::error::Error + Send + Sync>>> {
```

Comment:
Perhaps `impl Error` is fine here? It's not like `Box<dyn Error>` is any more useful from a type perspective.

Reply:
The `serve()` future that's spawned here returns `Result<(), Box<dyn Error + Send + Sync>>`, so that's just being forwarded here. I think there's an implicit `Sized` bound here too, so `impl Error` fails to compile.